### PR TITLE
add type definations for nodemark

### PR DESCRIPTION
**Pull request by beenotung**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/nodemark/package.json [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
+  "private": true,
+  "dependencies": {}
+}
```
Comment (peterblazejewicz) [inline types/nodemark/package.json:4]: this file serves no purpose, mind removing it?

File: types/nodemark/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,25 @@
+{
+    "compilerOptions": {
+        "esModuleInterop": true,
+        "module": "commonjs",
```
Comment (peterblazejewicz) [inline types/nodemark/tsconfig.json:4]:
```suggestion
        "module": "commonjs",
```
this is default (or set by user), when consuming packages, here not required, see:
https://github.com/DefinitelyTyped/DefinitelyTyped#esmoduleinteropallowsyntheticdefaultimports

File: types/nodemark/tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+        "lib": [
+            "dom",
```
Comment (peterblazejewicz) [inline types/nodemark/tsconfig.json:6]: why do you need this? if for `console.log`, just use TS lint assertions:
https://github.com/DefinitelyTyped/DefinitelyTyped#my-package-teststs

File: types/nodemark/index.d.ts [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+        sd(
+            precision?: number, // default to 0
+        ): number;
+
+        toString(
+            format?: Unit, // default to 'hz'
+        ): string;
+
+        // [util.inspect.custom]: () => string;
```
Comment (peterblazejewicz) [inline types/nodemark/index.d.ts:50]:
```suggestion
```
no dead code

File: types/nodemark/README.md [hunk 5]
```diff
@@ -0,0 +1,14 @@
+# @types/nodemark
```
Comment (peterblazejewicz) [inline types/nodemark/README.md:1]: why do you want dedicated license? btw, it's different from native package license, which is MIT

File: types/nodemark/nodemark-tests.ts [hunk 6]
```diff
@@ -0,0 +1,71 @@
+import benchmark from 'nodemark';
```
Comment (peterblazejewicz) [inline types/nodemark/nodemark-tests.ts:1]:
```suggestion
import benchmark = require('nodemark');
```

File: types/nodemark/README.md (hunk 5 above)
Comment (sheetalkamat) [reply types/nodemark/README.md:1]: Please remove this file.

**Review by sheetalkamat — CHANGES_REQUESTED** [review]
need to handle feedback from reviewers

File: types/nodemark/README.md (hunk 5 above)
Comment (beenotung) [reply types/nodemark/README.md:1]: I've no problem to remove the readme and different license.
But I wonder if this repo discourages adding type packages in different license?

File: types/nodemark/index.d.ts (hunk 4 above)
Comment (beenotung) [reply types/nodemark/index.d.ts:50]: I want to hint that field exists but not sure how to do that within the type system

File: types/nodemark/nodemark-tests.ts (hunk 6 above)
Comment (beenotung) [reply types/nodemark/nodemark-tests.ts]: adopted the suggestion

File: types/nodemark/package.json (hunk 1 above)
Comment (beenotung) [reply types/nodemark/package.json]: adopted the suggestion

File: types/nodemark/tsconfig.json (hunk 2 above)
Comment (beenotung) [reply types/nodemark/tsconfig.json]: agree, adopted the suggestion

File: types/nodemark/tsconfig.json (hunk 3 above)
Comment (beenotung) [reply types/nodemark/tsconfig.json:5]: removed 'dom' library from tsconfig

Comment (beenotung) [reply types/nodemark/tsconfig.json]: I manually declare the console instead, so now it works in CI (static analysis without running). And it also work when run locally with ts-node calling the underneath js package.

**Review by RyanCavanaugh — APPROVED** [review]

**Comment by beenotung** [comment]
Ready to merge

**Comment by beenotung** [comment]
This PR is merged 🎉
Thanks everyone involved

**Pull request merged**